### PR TITLE
Fix NetInfo.isConnected

### LINKED ISSUE #328: NetInfo isConnected event listener not firing
The below code should display a `connected: online` alert upon startup and `connected: offline` alert when e.g. WiFi is turned off. However, no alerts appear.

```
import React, {Component} from 'react';
import { AppRegistry, NetInfo, Modal, Text, View, TouchableHighlight } from 'react-native';
class rn_popup extends Component {
    render () {
          return (
                  <View style={{backgroundColor: 'rgba(0,0,0,0.4)'}}>
                    <Text>some text here</Text>
                    <Text>and more text here</Text>
                  </View>
          );
        }
}

function handleFirstConnectivityChange(isConnected) {
  alert('First change, isConnected: ' + (isConnected ? 'online' : 'offline'));
}
NetInfo.isConnected.addEventListener(
  'connectionChange',
  handleFirstConnectivityChange
);

NetInfo.getConnectionInfo().then((connectionInfo) => {
  alert('Initial, type: ' + connectionInfo.type + ', effectiveType: ' + connectionInfo.effectiveType);
});

AppRegistry.registerComponent('rn_popup', () => rn_popup);
```

### PR DESCRIPTION
**Pull request by siphiuel**
Fixes #328 

As it turns out, neither `QNetworkAccessManager::networkAccessible` nor `QNetworkConfigurationManager::isOnline` work properly when e.g. WiFi connection is disabled. These only check local interface availability, not remote host reachability. So they always return `true`.

Thus a QTimer solution periodically sending HEAD request to www.google.com is implemented. Not sure if this is good enough, but it works.

Notes: QTimer/QNetworkReply/QNetworkAccessManager heap management should be improved in this PR.

Also, hardcoded URL and timeout values might be declared as `const`.

### CODE REVIEW

File: ReactQt/runtime/src/netinfo.cpp [hunk 1]
```diff
@@ -12,31 +12,111 @@
...
         QObject::connect(bridge->networkAccessManager(),
                          &QNetworkAccessManager::networkAccessibleChanged,
                          [=](QNetworkAccessManager::NetworkAccessibility accessible) {
                              bridge->eventDispatcher()->sendDeviceEvent(
                                  "networkStatusDidChange",
-                                 QVariantList{QVariantMap{{"network_info", accessibleName.value(accessible)}}});
+                                 QVariantMap{{"connectionType", accessibleName.value(accessible)}});
                          });
+
+        timer = new QTimer();
```
Comment (MaxRis) [inline ReactQt/runtime/src/netinfo.cpp]: Having `timer` on stack should work fine here.

**Comment by vkjr** [comment]
@siphiuel, do you have clang-format turned on in QtCreator? `react-native-desktop` contains config for it to ensure code is properly styled. Here are screens with its settings:

screen shot 2018-09-20 at 5 37 35 pm

screen shot 2018-09-20 at 5 37 53 pm

Also there are few commented out lines of code. Probably they can be removed.

File: ReactQt/runtime/src/netinfo.cpp [hunk 2]
```diff
@@ -12,31 +12,111 @@
...
-    {QNetworkAccessManager::NotAccessible, "NotAccessible"},
-    {QNetworkAccessManager::Accessible, "Accessible"}};
+    {QNetworkAccessManager::UnknownAccessibility, "unknown"},
+    {QNetworkAccessManager::NotAccessible, "none"},
+    {QNetworkAccessManager::Accessible, "wifi"}};
 }
 
 class NetInfoPrivate {
 public:
     void monitorNetworkAccess() {
```
[the remark this answers is not in the document]
Comment (siphiuel) [reply ReactQt/runtime/src/netinfo.cpp]: Got it, done.

File: ReactQt/runtime/src/netinfo.cpp (hunk 1 above)
Comment (siphiuel) [reply ReactQt/runtime/src/netinfo.cpp]: Fixed.

**Comment by siphiuel** [comment]
@vkjr Ran `clang-format` on the file.

File: ReactQt/runtime/src/netinfo.cpp [hunk 3]
```diff
@@ -11,41 +11,111 @@
...
+        replyTimer.start();
+    });
+
+    requestTimer.start(NETWORK_REQUEST_TIMEOUT);
+}
 void NetInfo::getCurrentConnectivity(const ModuleInterface::ListArgumentBlock& resolve,
                                      const ModuleInterface::ListArgumentBlock& reject) {
     Q_UNUSED(reject);
     Q_D(NetInfo);
-    resolve(d->bridge,
```
Comment (MaxRis) [inline ReactQt/runtime/src/netinfo.cpp]: curios if that was removed, do we still need `NetInfo::getCurrentConnectivity` defined in general?

**Comment by vkjr** [comment]
@siphiuel, should we merge this?

**Comment by siphiuel** [comment]
@vkjr Still in QA

**Pull request merged**